### PR TITLE
decode binary strings in nxdl_utils.py

### PR DESCRIPTION
**Pull request by rettigl**

### CODE REVIEW

File: dev_tools/utils/nxdl_utils.py [hunk 1]
```diff
@@ -765,16 +765,28 @@ def get_best_child(nxdl_elem, hdf_node, hdf_name, hdf_class_name, nexus_type):
...
+        if (
+            "name" in nxdl_elem.attrib.keys()
+            and nxdl_elem.attrib["name"] == "NXdata"
+            and hdf_node is not None
+            and hdf_node.parent is not None
+            and hdf_node.parent.attrs.get("NX_class") == "NXdata"
+        ):
+            (fnd_child, fit) = get_best_nxdata_child(nxdl_elem, hdf_node, hdf_name)
+            if fnd_child is not None:
+                return (fnd_child, fit)
```
Comment (lukaspie) [inline dev_tools/utils/nxdl_utils.py:789]:
```suggestion
    if (
        "name" in nxdl_elem.attrib
        and nxdl_elem.attrib["name"] == "NXdata"
        and hdf_node is not None
        and hdf_node.parent is not None
    ):
        nx_class = hdf_node.parent.attrs.get("NX_class")
        if isinstance(nx_class, bytes):
            nx_class = nx_class.decode("UTF-8")
        if nx_class == "NXdata":
            (fnd_child, fit) = get_best_nxdata_child(nxdl_elem, hdf_node, hdf_name)
            if fnd_child is not None:
                return fnd_child, fit

```

**Review by lukaspie — APPROVED** [review]
I suspect that this byte array case is not checked anywhere in nxdl_utils, but for now it is fine to only check it here (if this is the part where it breaks).

**Review by sanbrock — APPROVED** [review]
LGTM

**Comment by mkuehbach** [comment]
https://github.com/FAIRmat-NFDI/pynxtools-em/blob/main/src/pynxtools_em/utils/hfive_utils.py L66 following is a generic implementation dealing with the  12 different string formatting strategies in HDF5

**Pull request merged**